peps/pep-0750.rst

```diff
@@ -718,88 +561,224 @@ interpolations, are as follows:
+
+Code that processes templates can parse the template string into intermediate
+representations, like an AST. We expect that many template processing libraries
+will use this approach.
+
+For instance, rather than returning a ``str``, our theoretical ``html()`` function
+(see the `Motivation`_ section) could return an ``HTMLElement`` defined in the 
+same package:
+
+.. code-block:: python
+
+    def html(template: Template) -> HTMLElement:
+        ...
+
+Calling ``str(element)`` would then render the HTML but, in the meantime, the
+``HTMLElement`` could be manipulated in a variety of ways.
+
+
+Context-sensitive Processing of Interpolations
+----------------------------------------------
+
+Continuing with our hypothetical ``html()`` function, it could be made 
+context-sensitive. Interpolations could be processed differently depending 
+on where they appear in the template. 
+
+For example, our ``html()`` function could support multiple kinds of
+interpolations:
+
+.. code-block:: python
+
+    attributes = {"id": "main"}
+    trade = "shrubbery"
+    content = "hello"
+    template = t"<div {attributes} data-trade={trade}>{content}</div>"
+    assert str(html(template)) == '<div id="main" data-trade="shrubbery">hello</div>'
+
+Because the ``{attributes}`` interpolation occurs in the context of an HTML tag,
+and because there is no corresponding attribute name, it is treated as a dictionary
+of attributes. The ``{trade}`` interpolation is treated as a simple string value
+and is quoted before inclusion in the final string. The ``{content}`` 
+interpolation is treated as potentially unsafe content and is escaped before 
+inclusion in the final string.
+
+
+Nested Template Strings
+-----------------------
+
+Going a step further with our ``html()`` function, we could support nested
+template strings. This would allow for more complex HTML structures to be
+built up from simpler templates:
+
+.. code-block:: python
+
+    name = "World"
+    content = html(t"<p>Hello {name}</p>")
+    template = t"<div>{content}</div>"
+    assert str(html(template)) == '<div><p>Hello World</p></div>'
+
+Because the ``{content}`` interpolation is an ``HTMLElement`` instance, it does
+not need to be escaped before inclusion in the final string.
+
+One could imagine a nice simplification: if the ``html()`` function is passed
+a ``Template`` instance, it could automatically process it as an ``HTMLElement``
+by recursively calling itself on the nested template.
+
+
+Approaches to Lazy Evaluation
+-----------------------------
+
+Like f-strings, interpolations in t-string literals are eagerly evaluated. However,
+there are cases where lazy evaluation may be desirable.
+
+If a single interpolation is expensive to evaluate, it can be explicitly wrapped
+in a ``lambda`` in the template string literal:
+
+.. code-block:: python
+
+    name = "World"
+    template = t"Hello {lambda: name}"
+    assert template.args[1].value() == "World"
```

Comment:
Perhaps emphasize the point with a line before this using:

assert callable(template.args[1].value)

Reply:
I like that. Added.